### PR TITLE
breakout feeding types

### PR DESCRIPTION
**Pull request by alzyee**
There are some bugs that this change appears to create.
* on mobile it puts the legend over the graph
* on mobile it makes the trend very narrow 

I tried adding this code to the layout but on mobile it overlaps the x axis labels
        `'legend':{
            'orientation':'h'
        }`

It might be better to just not have a label. 
in reference to #383

### CODE REVIEW

File: reports/graphs/feeding_amounts.py [hunk 1]
```diff
@@ -14,23 +14,57 @@ def feeding_amounts(instances):
     :param instances: a QuerySet of Feeding instances.
     :returns: a tuple of the the graph's html and javascript.
     """
-    totals = {}
+    feeding_types = [
```
Comment (cdubz) [inline reports/graphs/feeding_amounts.py:17]: You can use model metadata to get these types directly from the model (after its imported). Some variation of `feeding_types = Feeding._meta.get_field('type').choices` should do the trick.

File: reports/graphs/feeding_amounts.py [hunk 2]
```diff
@@ -14,23 +14,57 @@ def feeding_amounts(instances):
...
+            'breast milk',
+            'formula',
+            'fortified breast milk',
+            'solid food',
+        ]
+    feeding_types_desc = [
```
Comment (cdubz) [inline reports/graphs/feeding_amounts.py:23]: See previous comment re: model field metadata. These descriptive names will come from the same place.

File: reports/graphs/feeding_amounts.py [hunk 3]
```diff
@@ -14,23 +14,57 @@ def feeding_amounts(instances):
...
+    for i in range(total_idx):
+        amounts_array.append([round(a, 2) for a in totals_list[i].values()])
+
+    traces = []
+    for i in range(total_idx-1):
+        traces.append(go.Bar(
+            name=feeding_types_desc[i],
+            x=list(totals_list[total_idx-1].keys()),
+            y=amounts_array[i],
+            text=amounts_array[i]
```
Comment (cdubz) [inline reports/graphs/feeding_amounts.py:55]: On hover right now we get too much information -- date, amount again, and type. Ideally we just have type.

image

I'm not 100% sure how to set it up that way -- take a look at https://plotly.com/python/hover-text-and-formatting/ and if ya get stuck I can take a closer look.

File: reports/graphs/feeding_amounts.py [hunk 4]
```diff
@@ -14,23 +14,57 @@ def feeding_amounts(instances):
...
+
```
Comment (cdubz) [inline reports/graphs/feeding_amounts.py:56]: Remove this empty space.

File: reports/graphs/feeding_amounts.py [hunk 5]
```diff
@@ -14,23 +14,57 @@ def feeding_amounts(instances):
...
         hoverinfo='text',
         textposition='outside',
-        text=amounts
-    )
+        text=amounts_array[total_idx-1],
+        showlegend=False
+    ))
 
     layout_args = utils.default_graph_layout_options()
     layout_args['title'] = _('<b>Total Feeding Amounts</b>')
```
Comment (cdubz) [inline reports/graphs/feeding_amounts.py:70]: Lets update the title of this graph to "Total Feeding Amount by Type". We'll need to do this elsewhere as well.

**Review by cdubz — CHANGES_REQUESTED** [review]
This a great start! A few change requests here. Let me know if you need any help with these changes.

File: reports/graphs/feeding_amounts.py (hunk 5 above)
Comment (alzyee) [reply reports/graphs/feeding_amounts.py]: I changed as suggested but it seems like this should somehow come from the translations. I'm not sure how to make that happen.

**Comment by alzyee** [comment]
I think I addressed all of your comments

File: reports/graphs/feeding_amounts.py (hunk 5 above)
Comment (cdubz) [reply reports/graphs/feeding_amounts.py]: @alzyee yep no worries -- that is what `_` does. It's a weird Python convention: https://github.com/babybuddy/babybuddy/blob/master/reports/graphs/feeding_amounts.py#L3. Translations will need to be updated to pick up the change but I'll do that separately.

**Review by cdubz — APPROVED** [review]
Looks great! Thanks, @alzyee.

**Pull request merged**